astrokat/targets.py

```diff
@@ -0,0 +1,436 @@
+            raise RuntimeError('Ill defined target, require: x and y')
+
+    target = "{}, {} {}, {}, {}, {}".format(name,
+                                            ctag,
+                                            tags,
+                                            x, y,
+                                            flux_model)
+    return name, target
+
+
+def build_target(target_dict, timestamp=None):
+    """Restructure dictionary into target defined recarray for observation"""
+    # When unpacking, katpoint's naming convention will be to use the first
+    # name, or the name with the '*' if given. This unpacking mimics that
+    # expected behaviour to ensure the target can be easily called by name
+    ctag = target_dict["coord"][0].strip()
+    x = target_dict["coord"][1].split()[0].strip()
+    y = target_dict["coord"][1].split()[1].strip()
+    [name_list, katpoint_tgt] = katpoint_target(name=target_dict["name"],
+                                                ctag=ctag,
+                                                x=x, y=y,
+                                                tags=target_dict["tags"],
+                                                flux_model=target_dict["flux_model"],
+                                                )
+    name_list = [name.strip() for name in name_list.split("|")]
+    prefered_name = list(filter(lambda x: x.startswith("*"), name_list))
+    if prefered_name:
+        target_name = prefered_name[0][1:]
+    else:
+        target_name = name_list[0]
+
+    # process observation keywords from YAML input
+    if "duration" in target_dict.keys():
+        duration = float(target_dict["duration"])
+    else:
+        duration = np.nan
+    if "cadence" in target_dict.keys():
+        cadence = float(target_dict["cadence"])
+    else:
+        cadence = -1  # default is to observe without cadence
+    if "obs_type" in target_dict.keys():
+        obs_type = target_dict["obs_type"]
+    else:
+        obs_type = "track"  # assume tracking a target
+    if "noise_diode" in target_dict.keys():
+        nd = target_dict["noise_diode"]
+    else:
+        nd = None
+    if "last_observed" in target_dict.keys():
+        last_observed = target_dict["last_observed"]
+    else:
+        last_observed = None
+    if "obs_cntr" in target_dict.keys():
+        obs_cntr = target_dict["obs_cntr"]
+    else:
+        obs_cntr = 0
+
+    return (target_name,
+            target_dict["tags"],
+            katpoint_tgt,
+            duration,
+            cadence,
+            target_dict["flux_model"],
+            obs_type,
+            nd,
+            last_observed,
+            obs_cntr
+            )
+
+
+def read(target_items, timestamp=None):
+    """Read targets info.
+
+    Unpack targets target items to a katpoint compatible format
+    Update all targets to have celestial (Ra, Dec) coordinates
+
+    """
+    ntargets = len(target_items)
+    target_list = np.recarray(ntargets, dtype=tgt_desc)
```

Comment:
I can understand that the use of the word `list` may  cause some confusion, since it is not a list as in the techinical sense, but rather and array of records.
So I am going to suggest again simply being explicit and updating it to: `target_list` -> `target_recarray`